pkg/cmd/prebuild/update.go

```diff
@@ -43,108 +52,131 @@ var prebuildUpdateCmd = &cobra.Command{
+				return errors.New("retention must be a number")
 			}
-		}
 
-		retention, err := strconv.Atoi(prebuildAddView.Retention)
-		if err != nil {
-			return errors.New("retention must be a number")
-		}
-
-		newPrebuild := apiclient.CreatePrebuildDTO{
-			Id:        &prebuild.Id,
-			Branch:    &prebuildAddView.Branch,
-			Retention: int32(retention),
-		}
-
-		if commitInterval != 0 {
-			newPrebuild.CommitInterval = util.Pointer(int32(commitInterval))
-		}
-
-		if len(prebuildAddView.TriggerFiles) > 0 {
-			newPrebuild.TriggerFiles = prebuildAddView.TriggerFiles
-		}
+			newPrebuild := apiclient.CreatePrebuildDTO{
+				Id:        &prebuild.Id,
+				Branch:    &prebuildAddView.Branch,
+				Retention: int32(retention),
+			}
 
-		prebuildId, res, err := apiClient.PrebuildAPI.SetPrebuild(ctx, prebuildAddView.ProjectConfigName).Prebuild(newPrebuild).Execute()
-		if err != nil {
-			return apiclient_util.HandleErrorResponse(res, err)
-		}
+			if commitInterval != 0 {
+				newPrebuild.CommitInterval = util.Pointer(int32(commitInterval))
+			}
 
-		views.RenderInfoMessage("Prebuild updated successfully")
+			if len(prebuildAddView.TriggerFiles) > 0 {
+				newPrebuild.TriggerFiles = prebuildAddView.TriggerFiles
+			}
 
-		if prebuildAddView.RunBuildOnAdd {
-			projectConfig, res, err := apiClient.ProjectConfigAPI.GetProjectConfig(ctx, prebuildAddView.ProjectConfigName).Execute()
+			prebuildId, res, err := apiClient.PrebuildAPI.SetPrebuild(ctx, prebuildAddView.ProjectConfigName).Prebuild(newPrebuild).Execute()
 			if err != nil {
 				return apiclient_util.HandleErrorResponse(res, err)
 			}
 
-			buildId, err := build.CreateBuild(apiClient, projectConfig, *newPrebuild.Branch, &prebuildId)
-			if err != nil {
-				return err
-			}
+			views.RenderInfoMessage("Prebuild updated successfully")
+
+			if prebuildAddView.RunBuildOnAdd {
+				projectConfig, res, err := apiClient.ProjectConfigAPI.GetProjectConfig(ctx, prebuildAddView.ProjectConfigName).Execute()
+				if err != nil {
+					return apiclient_util.HandleErrorResponse(res, err)
+				}
 
-			views.RenderViewBuildLogsMessage(buildId)
+				buildId, err := build.CreateBuild(apiClient, projectConfig, *newPrebuild.Branch, &prebuildId)
+				if err != nil {
+					return err
+				}
+
+				views.RenderViewBuildLogsMessage(buildId)
+			}
 		}
+
 		return nil
 	},
 }
 
-var runOnUpdateFlag bool
-
 func init() {
+	prebuildUpdateCmd.Flags().StringVar(&branchFlag, "branch", "", "Git branch for the prebuild")
+	prebuildUpdateCmd.Flags().IntVar(&retentionFlag, "retention", 0, "Retention period for the prebuild")
+	prebuildUpdateCmd.Flags().IntVar(&commitIntervalFlag, "commit-interval", 0, "Commit interval for the prebuild")
+	prebuildUpdateCmd.Flags().StringSliceVar(&triggerFilesFlag, "trigger-files", nil, "Files that trigger the prebuild")
+	prebuildUpdateCmd.Flags().StringVar(&projectConfigFlag, "project-config", "", "Project configuration name") // New flag
```

Comment:
Same comment as above, `project-config` should be an arg.

Reply:
same updated

Reply:
`project-config` is still a flag here

Reply:
Updated and removed it, currenlty it takes the value from the arguments.